Addition of `sort_adj` to sort an rank 1 array in the same order as an input array (#849)
fortran-lang/stdlib: doc/specs/stdlib_sorting.md

* `ORD_SORT` is intended to sort simple arrays of intrinsic data
  that have significant sections that were partially ordered before
  the sort;
* `SORT_ADJOINT` is based on `ORD_SORT`, but in addition to sorting the

>   input array, it returns a related array re-ordered in the

Reviewer: ```suggestion
  input array, it re-orders a second array of the same size
```

>   same way;

Reviewer: ```suggestion
  according to the same permutations;
```

Arrays can be also sorted in a decreasing order by providing the argument `reverse
= .true.`.

#### The `SORT_ADJOINT` subroutine

The `SORT` and `ORD_SORT` subroutines can sort rank 1 isolated
arrays of intrinsic types, but do nothing for the coordinated sorting
of related data, e.g., a related rank 1 array. Therefore the module
provides a subroutine, `SORT_ADJOINT`, that re-order such a rank 1 array

> in the same way as the input array based on the `ORD_SORT` algorithm,

Author: ```suggestion
according to the same permutations as for the input array based on the `ORD_SORT` algorithm,
```

{!example/sorting/example_radix_sort.f90!}
```

> #### `sort_adjoint` - sorts an associated array in the same way as the input array, while also sorting the array.

Author: ```suggestion
#### `sort_adjoint` - sorts an associated array 
according to the same permutations as for the input array.
```

##### Status

Experimental

##### Description

Returns the input `array` sorted in the direction requested while
retaining order stability, and an associated array whose elements are

> sorted in the same way as the input `array`.

Author: ```suggestion
sorted according to the same permutations as for the input `array`.
```
